examples/cpp_api/using_tiledb_stats.cc

```diff
@@ -89,6 +89,34 @@ void read_array() {
   Stats::enable();
   query.submit();
   Stats::dump(stdout);
+
+  // Check stats.
+  std::string stats;
+  Stats::dump(&stats);
+  if (stats.find("\"Context.StorageManager.subSubarray.add_range\": 2") ==
+      std::string::npos) {
+    throw std::logic_error("Invalid counter for add_range");
+  }
+
+  // Ensure additional calls to Query::submit have no effect on the stats
+  query.submit();
+  query.submit();
+  Stats::dump(&stats);
+  if (stats.find("\"Context.StorageManager.subSubarray.add_range\": 2") ==
+      std::string::npos) {
+    throw std::logic_error("Invalid counter for add_range");
+  }
+
+  // Invoke add_range and check the stats again.
+  // #TODO Update After removal of deprecated Query::add_range
+  query.add_range(0, (uint32_t)0, (uint32_t)3);
+  query.add_range(1, (uint32_t)0, (uint32_t)3);
+  query.submit();
+  Stats::dump(&stats);
+  if (stats.find("\"Context.StorageManager.subSubarray.add_range\": 4") ==
```

Comment:
subSubarray seems weird? Any clue what leads to this?

Reply:
I thought this was odd at first too but it's how the `Subarray::stats_` member variable is initialized here: https://github.com/TileDB-Inc/TileDB/blob/21f4c1a3b5231db03242f133fde83aeed78f012c/tiledb/sm/subarray/subarray.cc#L115-L119